frontend/src/tests/instructor-permission-test.js

```diff
@@ -516,51 +469,254 @@ export function instructorsPermissionTests(api) {
+                assignment_id: assignment.id,
+                duties: [
+                    {
+                        order: 1,
+                        hours: 40,
+                        description: "marking:Test Marking",
+                    },
+                    {
+                        order: 2,
+                        hours: 30,
+                        description: "other:Additional duties",
+                    },
+                ],
+                approved_date: new Date().toISOString(),
+                accepted_date: "2020-09-02T00:00:00.000Z",
+                revised_date: "2020-09-01T00:00:00.000Z",
+                emailed_date: "2020-08-28T00:00:00.000Z",
+                signature: "Harry Potter",
+            };
+            resp = await apiPOST(`/instructor/ddahs`, ddahWithRestrictedFields);
+            expect(resp).toHaveStatus("success");
+            expect(resp.payload).toEqual(expect.objectContaining(originalDdah));
+
+            resp = await apiGET(`/instructor/sessions/${session.id}/ddahs`);
+            expect(resp).toHaveStatus("success");
+            const newDdah = resp.payload.find(
+                (ddah) => ddah.assignment_id === assignment.id
+            );
+            expect(newDdah).toEqual(expect.objectContaining(originalDdah));
+        }); // still a success, but does not change the underlying data
+
+        it("cannot create a Ddah for an assignment not associated with self", async () => {
+            const ddahToCreate = {
+                // This assignment has no DDAH initially
+                assignment_id: foreignAssignment.id,
+                duties: [
+                    {
+                        order: 1,
+                        hours: 20,
+                        description: "other:Facilitating workshops",
+                    },
+                ],
+            };
+            let resp = await apiPOST(`/instructor/ddahs`, ddahToCreate);
+            expect(resp).toHaveStatus("error");
+            expect(resp.message).toStrictEqual(
+                "Cannot create a DDAH without an assignment_id OR improper permissions to access/create a DDAH"
+            );
+        });
+
+        it("cannot update a Ddah for an assignment not associated with self", async () => {
+            // First we add the DDAH to the assignment
+            // associated with another instructor
+            await restoreDefaultUser();
+            const ddahToCreate = {
+                assignment_id: foreignAssignment.id,
+                duties: [
+                    {
+                        order: 1,
+                        hours: 30,
+                        description: "contact:Practice",
+                    },
+                ],
+            };
+            let resp = await apiPOST(`/admin/ddahs`, ddahToCreate);
+            expect(resp).toHaveStatus("success");
+            expect(resp.payload.duties).toStrictEqual(ddahToCreate.duties);
```

Comment:
We've already done this check in another test, so you can just assume it's working.

Reply:
Ok, will delete. Out of curiosity, wouldn't that be consistent with testing our assumptions (like Mert suggested with payload to be defined before `.find` it)?